src/ripple/overlay/impl/TrafficCount.h

```diff
@@ -122,9 +183,8 @@ class TrafficCount
         return ret;
```

Comment:
I can't leave a comment on lines 179-180, but I'm pretty sure `std::piecewise_construct` and `std::forward_as_tuple` are not required here because we're only passing one argument and there are no ambiguous constructors. I'm guessing someone just copied the pattern on cppreference.com without thinking too much.

Further, now that we have `std::atomic`s in `TrafficStats`, do we want to construct copies of those for each category, or are raw `std::uint64_t`s good enough? I don't know much about the cost of `std::atomic`.

I'm guessing using `std::atomic` implies this class wants to be thread-safe, but there is a race condition here: interleaving the copies of related values (e.g. bytes in + messages in) on line 180 with updates to those values in lines 150-156. So there is a small chance that a small discrepancy can arise, but since these are just diagnostics, I'm guessing that potential error can be tolerated.

Reply:
They were `std::atomic` before (behind a typedef), so the behavior should be unchanged, just changed from `unsigned long` to `uint64_t`

Reply:
Sure, but that doesn't change the point. Anyone requesting a copy probably doesn't need atomics. `TrafficStats` could be a template instantiated for `uint64_t` and `std::atomic<uint64_t>`. `std::atomic<T>` has an `operator T` so you can even write a polymorphic copy constructor.

Reply:
I think this is a fair point, @thejohnfreeman. I'd rather not make this change at this time, but it's something we should look at for the next round of improvements (and there are a few others too) we do.